File: Programmers/Level2/올바른 괄호.py
from collections import deque
def solution(s):
    answer = True
    
    q = deque()
    for i in range(len(s)):
        if s[i]==')' and not q:
            return False
        elif s[i]==')':
            q.pop()
        else:
            q.append(s[i])
    if q:
        return False
    return True

File: Programmers/Level2/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {

        Stack<Character> st = new Stack<>();
        for (int i=0;i<s.length();i++) {
            char ch = s.charAt(i);
            if (ch == '(') {
                st.push(ch);
            }
            else if (ch == ')' && st.empty()) {
                return false;
            }
            else {
                st.pop();
            }
        }
        if (st.empty()) {
            return true;
        }
        return false;
    }
}